src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  static updateAlive() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    Animal.alive.push(this);
+  }
+  bite(victim) {
+    if (victim instanceof Herbivore && !victim.hidden) {
+      victim.health -= 50;
+      Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
```

Comment:
The line `Animal.alive = Animal.alive.filter((animal) => animal.health > 0);` is redundant here because the `updateAlive` method already performs this operation. Consider removing this line to avoid unnecessary repetition.

Reply:
> The line `Animal.alive = Animal.alive.filter((animal) => animal.health > 0);` is redundant here because the `updateAlive` method already performs this operation. Consider removing this line to avoid unnecessary repetition.

Good point